More adjustment suggestions (#25)
OR13/diem-draft-charter: diem-unified-charter.md

Emblems may be observed by validators without the knowledge of the bearer displaying the emblem, or may be presented to a specific validator upon request.
Cryptographic verification can be optional as long as the emblem can be correctly interpreted.
Therefore, to be effective, the semantics of an emblem must be well known, easily recognizable, and distinguishable from other emblems.

Digital emblems can be unsigned, self-signed, or signed by a trust anchor.
Which attributes an emblem contains, and how a digital emblem is secured and presented impacts how a validator interprets and trusts the assertions made within the emblem.

# Initial Scope

The DIEM WG will ensure maximum reuse of work for emblem representation and binding. 
A discovery mechanism for the initial work will only be specified by this group after the initial emblem binding protocol is completed (see the Deliverables section below).

The working group is initially limited to specifying discovery mechanisms that rely on digital communication such as the use of DNS or well-known locations on a host identified by a hostname or IP address.

> The working group working group must not propose discovery mechanisms or binding solutions that enable validation to be distinguishable from currently existing interactions with the emblem bearer. 

Author: ```suggestion
Only discovery mechanisms where the validation remains unknown to the bearer of the emblem are considered in the initial scope.
```

Author: I changed the word validator to validation

Reviewer: How about "The working group will develop discovery mechanisms where by default the validator and validation remain unknown to the bearer of the emblem."

Author: The default implies you want interactive validation / verification / presentation in scope.
If thats true... just say that, and we can gauge consensus.

# Introduction

An emblem is a device, symbol, or figure adopted and used as an identifying mark.
In culture, emblems such as a flag, badge or coat of arms communicate group identity.
Such emblems are often comprised of a set of attributes each with symbolic meaning.
In speech, emblems are specific nonverbal gestures or signals that have a direct verbal translation and are widely understood within a particular culture or community.

> Under International Humanitarian Law (IHL), the red cross, red crescent, and red crystal emblems are symbols of protection.

Reviewer: ```suggestion
Under International Humanitarian Law (IHL), the Red Cross, Red Crescent, and Red Crystal emblems are symbols of protection.
```

Similarly, the "blue shield" is another IHL emblem that marks cultural property under special protection.
ISO 7010 defines a set of emblems that can be used to identify hazards, these include the skull and crossbones for toxic material, the ionizing radiation symbol and the biological hazard symbol.
Today often these emblems/symbols require a sense of sight, or touch to become known to the receiver.

There is a need to sense emblems/symbols through digital communication channels.

Digital emblems extend the range of identifying marks from the physical (visual and tactile) to the digital realm.
The presence of a digital emblem represents a new signal available to cyber operators. 

The DIEM WG will define an architecture and discovery mechanism enabling digital emblems to be presented and validated across applications and platforms in a cohesive way.

# Architectural Consideration

"To bear an emblem" means to present or display an identifying mark. 
The entity that bears the emblem is respectively the bearer or emblem holder. 

> This can be a separate entity from the creator or original designer of the emblem.

Reviewer: ```suggestion
This is often a separate entity from the creator or original designer of the emblem.
```

"To validate an emblem" means to confirm the authenticity or legitimacy of a particular symbol or design, often by checking its details against a known standard or reference point. 
Emblems may be observed by validators without the knowledge of the bearer displaying the emblem, or may be presented to a specific validator upon request.

> Cryptographic verification can be optional as long as the emblem can be correctly interpreted.

Reviewer: Who has asked for optional verification? This sounds like the moral equivalent of the "evil bit".

Author: Its more that I don't want to get into the details of "verifier" vs "signature checker"... and we seem to have evidence that verification could come at the transport layer.... put another way, who said a new PKI should be mandatory?

Therefore, to be effective, the semantics of an emblem must be well known, easily recognizable, and distinguishable from other emblems.

Digital emblems can be borne or presented by entities that are analog (e.g., a vehicle) or digital (e.g., a web page), however, there needs to be a digital representation that is used for validation using digital communication.

> Digital emblems can be unsigned, self-signed, or signed by a trust anchor.

Reviewer: who wants unsigned emblems and why will that be "safe"?

Reviewer 2: Thinking out loud. The initial paragraphs mentions emblems to mark hazards. Unsigned, digital hazard symbols are not a priori useless. For example, they could inform the validator that "proceeding" poses a "danger."

Having said that, I don't see this coming up really. I parse this sentence "the charter does not require emblems to be signed." While I think that all emblems will be signed, I think it's fine that the charter doesn't enforce this.

Author: message vs transport security... also in an emergency setting up signing infrastructure might not be first priority.

Reviewer: Having worked in a few emergency settings, you rarely create or even configure a new anything. You have a tested system in the warehouse and you ship it. The signing infra would be ready and tested. So, I don't see emergency settings as a motivator to skip signing.

Which attributes an emblem contains, and how a digital emblem is secured and presented impacts how a validator interprets and trusts the assertions made within the emblem.

# Initial Scope

The DIEM WG will ensure maximum reuse of work for emblem representation and binding. 
A discovery mechanism for the initial work will only be specified by this group after the initial emblem binding protocol is completed (see the Deliverables section below).

> The working group is limited to specifying discovery mechanisms that rely on digital communication such as the use of DNS or well-known locations on a host identified by a hostname or IP address.

Reviewer: ```suggestion
The working group is initially limited to specifying discovery mechanisms that rely on digital communication such as the use of DNS or well-known locations on a host identified by a hostname or IP address.
```

Reviewer 3: I agree that re-adding the work "initially" here is important to make clear that the group could recharter after addressing this initial scope.

The working group working group must not propose discovery mechanisms or binding solutions that enable validation to be distinguishable from currently existing interactions with the emblem bearer. 

> The design of discovery mechanisms using proximity-based protocols such as QRCodes, NFC, or Bluetooth is out-of-scope.

Reviewer: This is not a deal breaker for me.

```suggestion
The design of discovery mechanisms using proximity-based protocols such as QRCodes, NFC, or Bluetooth will not be considered until the other deliverables have substantially progressed.
```

Reviewer 3: I think this should say that those may be considered after rechartering. Maybe it would be good to still add a sentencing saying that architecture however should those discovery mechanisms already into account?

Author: I prefer not to accept this proposed text.

Reviewer: @OR13 where do you see that work happening?

Author: In discussions with the group, I have heard no objections to DNS related binding / discovery.
I have heard repeated objections to proximity validation.
In the interest of getting a charter that has consensus, I am trying to cut where there is consensus.
If we keep holding onto parts where there is no consensus, we won't be able to push a new charter to the IESG for consideration of external review.

Its implied, but I will state it again, if the scope with consensus is completed, new scope can be added with a re-charter.

The working group will facilitate re-use of existing protocols and capabilities and ensure that existing standards are leveraged appropriately.
The working group will not produce any standards track generic serialization formats. 
The working group will not produce any standards track extensions to the DNS. 

> The working group will not develop novel security mechanisms, cryptographic primitives, or digital signature schemes. 

Reviewer 2: @OR13 These sentences are in the "initial scope" section. Was that intentional? Doesn't seem like the best place. I presume that these restrictions also apply after the initial scope was addressed.

Author: The charter scoping applies until the working group is closed, or a new charter is approved.
Given this is the first charter, everything is technically "initial scope".
However, folks seem to like attaching the word initial to scope to signal that it might expand in the future.
These requirements could be dropped (or expanded) with a recharter.

Reviewer: Felix said:
> The requirement of "not revealing private information" is the wrong requirement. It is neither covert inspection nor does it reflect the requirement you mention in your comment (informing the bearer about validation). I find that Orie accurately phrased covert inspection.

I believe that I accurately conveyed the requirement. Please see my inline comment above. It is possible that Felix is using a narrow sense of "private" and I am using a broad sense.

# Introduction

An emblem is a device, symbol, or figure adopted and used as an identifying mark.
As such, emblems provide a mechanism for conveying a set of attributes bound to an asset. 

> An emblem may be presented to a validator and attributes are asserted by an issuer.

Author: Too much detail in introduction, we  introduce roles in the architecture considerations section.
I also feel this is somewhat redundant given the current text

Reviewer 3: Yes, I also felt this doesn't belong here but in the architecture section. However, I don't think you moved the text there, did you? That fine, but I wanted to point out that we are now loosing the bit on "attributes". However, maybe this is not important or actually too detailed for the charter, anyway?

Today often these emblems/symbols require a sense of sight, or touch to become known to the receiver.

There is a need to sense emblems/symbols through digital communication channels.

> The DIEM WG extends the range of identifying marks from the physical (visual and tactile) to the digital realm.

Reviewer 3: ```suggestion
Digital emblems extend the range of identifying marks from the physical (visual and tactile) to the digital realm.
```

The presence of a digital emblem represents a new signal available to cyber operators. 

> The DIEM WG will define an architecture and discovery mechanism enabling digital emblems to be presented and validated across applications and platforms in a cohesive way.

Author: I collapsed deliverables here, this was based on feedback regarding the relationship between "binding and discovery".

# Architectural Consideration

> "To issue or produce" an emblem, is to design the identifying mark and its associated attributes.

Author: Instead of using issuer in the intro, i tried to define the concept here, so that the sentence below about "holders" not being issuers would have context...

This text still suffers from the abstract nature of "emblem creator (standardizing body)" vs "emblem issuer (service provider)"

Reviewer 3: ```suggestion
"To issue or produce an emblem", is to create the identifying mark and its associated attributes.
```

Reviewer 3: Actually now I'm confused here. I thought the issuer of a digital emblem is an entity that creates a specific emblem for a specific asset to assert certain attributes to that asset. e.g this could be credentials that then can be retrieved through the DNS. But the issuer wouldn't "design" it because the format of the emblem is based on some standard...

Author: exactly what I am struggling with.

Author: https://github.com/OR13/diem-draft-charter/pull/25/commits/363784c65deb7ac71e7ad56ece12011aa056b855

Author: I added attributes back explicitly in: https://github.com/OR13/diem-draft-charter/pull/25/commits/8a5dd1ad0a93f10fe8f633ed0e0f04ea1d3a4e35

Reviewer: It sounds like the problem is the difference between "a type of emblem" and "an instance of an emblem".

Author: Yes I think so, type is set of required attributes (emblem standardizer), instance is an instantiation of a type (emblem issuer)... but I hope we can avoid adding emblem type back.

> The emblem issuer's role is to ensure that the identifying mark is easy to recognize and well understood.

Reviewer 3: not sure this sentence is useful. Also the issues, as it was written above, is rather then one that asserts certain attributes to an asset by providing the (instant of an) emblem....? the standards body defining the emblem needs to ensure it's well recognisable.

Author: let me see if I can eliminate the emblem issuer from the charter text

"To bear an emblem" means to present or display an identifying mark. 
The entity that bears the emblem is respectively the bearer or emblem holder. 
This can be a separate entity from the emblem issuer.

"To validate an emblem" means to confirm the authenticity or legitimacy of a particular symbol or design, often by checking its details against a known standard or reference point. 

> Emblems may be observed by validators without the knowledge of the bearer displaying the emblem, or may be presented to a specific validator upon request.

Author: This was a comment from @felixlinker  on a previous PR

Cryptographic verification can be optional as long as the emblem can be correctly interpreted.
Therefore, to be effective, the semantics of an emblem must be well known, easily recognizable, and distinguishable from other emblems.

> Digital emblems can be borne or presented by entities that are analog (e.g., a vehicle) or digital (e.g., a web page), however, there needs to be a digital representation that is used for validation using digital communication.

Author: I reworked this... Its still vague in the sense of "are the people in the vehicle marked" or "are adjacent services published in the web page marked", but thats too much detail for a charter probably.

Reviewer 2: ```suggestion
```

I'd again suggest removing this sentence. I think it raises more questions than it answers. Also: (A) We don't use the analog/digital distinction anymore in the charter. (B) This distinction was suggested as complete, i.e., there is no third type of entities that even could bear emblems. Thus, the sentence is vacuous. (As is "Entities are X or not X.) (C) The part that "there needs to be a representation for digital communication" is covered elsewhere in the charter already too.

Reviewer 3: Why introduce a new verb ("borne") here?

However, as I also said previously, I'd be okay to remove this sentence.

> Digital emblems can be unsigned, self-signed, or signed by the emblem issuer.

Author: Attestation is a dangerous word : ) I hope this is clearer, it also related signing to the emblem issuer.

Reviewer 3: fine to remove this sentence.

> How a digital emblem is secured and presented impacts how a validator interprets and trusts the assertions made within the emblem.

Reviewer 3: this is good

# Initial Scope

The DIEM WG will ensure maximum reuse of work for emblem representation and binding. 
A discovery mechanism for the initial work will only be specified by this group after the initial emblem binding protocol is completed (see the Deliverables section below).

The working group is limited to specifying discovery mechanisms that rely on digital communication such as the use of DNS or well-known locations on a host identified by a hostname or IP address.

> The working group working group must not propose discovery mechanisms or binding solutions that enable validation to be distinguishable from currently existing interactions with the emblem bearer. 

Author: This is a rewording of the "covert inspection" requirement.
The current text is not really constraining, since it says "address the challenge".
I hope I have captured the essence of the original intent here.

Reviewer: I **strongly** disagree with the wording above. Some emblem use cases require the bearer to know they are being queried. We should be developing tools that work in both use cases on an emblem-by-emblem basis. Hopefully my text below is acceptable.

```suggestion
The working group will develop discovery mechanisms and/or binding solutions that, by default, do not reveal private information about the validation or validator to the emblem bearer. 
```

Reviewer 2: The requirement of "not revealing private information" is the wrong requirement. It is neither covert inspection nor does it reflect the requirement you mention in your comment (informing the bearer about validation). I find that Orie accurately phrased covert inspection.

Informing bearers about emblem validation and covert inspection are contradictory. I think it's too easy to say that "we should simply provide designs that work for both requirements." This sets us up for failure.

I suggest saying more explicitly that all restrictions in this paragraph apply to the *initial* scope, not to the WG in general, and keep covert inspection as part of the initial scope. Should we succeed with the initial scope, it seems easy to extend designs providing covert inspection to inform the bearer of validation (you simply hide what could be covertly discovered before behind an interactive protocol). As the WG will "ensure maximum reuse," this should be very much possible!

Reviewer 3: I think "must not propose" is not a great wording as everybody in a group is of course free to _propose_ things even when out of scope.

I think I would prefer something that is closer to what was described in the initial scope previously like:

> Only discovery mechanisms where the validator remains unknown to the bearer of the emblem are considered in the initial scope.

Reviewer 2: I like that phrasing!

Author: I proposed that exact text here: https://github.com/OR13/diem-draft-charter/pull/25/files#r1959836310

But I think its a mistake because the requirement in not about the validator, it is about monitoring validation.

Its not about the validator remaining unknown, its about validation being "for free" or "part of an existing high traffic flow with the existing infrastructure".

The emblem bearer expects validation, they hope there will be validators... the fact that validation is indistinguishable from normal traffic increases the chance that validation will happen by the right validators.

Reviewer 2: Yes, you're right. The phrasing suggested by your edit is better.

Reviewer 2: Would it maybe be more simple to say:

> Only discovery and binding mechanisms where validation remains undetectable as such are considered in the initial scope.

Reviewer 2: PS: Added "and binding" in my phrasing. I think that's important as binding will probably also be validated.

Reviewer: I have two problems here. The first is that this requirement is needed for a small subset of emblems, and is the opposite of what we want for another subset. I think most emblems don't care. That's why I propose the "by default" language here.

Then there is how we describe the requirement. While Mirje's phrasing is better, this is not focusing on the private data being revealed (Felix, think of the word "sensitive"). Even if the identity of the specific validator is unknown, other details of the validation (ex:timing) may be sensitive. That is why I used "..about the validation or validator..."

Reviewer 4: > Would it maybe be more simple to say:
> 
> > Only discovery and binding mechanisms where validation remains undetectable as such are considered in the initial scope.

I am supportive of this phrasing as an improvement and ack of raised points.

Rohan, I think this aligns the scoping with what Brian and I penned but the IESG couldn't support because it was too general (that digital emblems are broadly inclusive of both scenarios, but a new WG will focus only on one at first). This case is the one with more novel technical requirements that make it unique and therefore stricter design. Knowing the latest IESG and IAB feedback is to make the charter tighter and more concrete, I think this is the correct line to draw; getting _more_ general is the wrong direction to go for initial chartering.

   The architecture must not assume any specific serialization formats or securing and discovery mechanisms.

3. A protocol specification describing the binding of emblems to entities (Proposed Standard or Experimental):
   After the architecture and data model are finished, the working group will specify a single concrete serialization for digital emblems and emblems types for one or two of the initial use cases. 
3. A protocol specification describing the discovery of digital emblems (Proposed Standard or Experimental):
   After the architecture and data model are finished, the working group will specify a binding and discovery mechanism addressing one or two of the initial use cases. 
   This specification may describe a mandatory to implement securing mechanism. 
   If a securing mechanism is described, at least one mandatory to implement cryptographic algorithm which is already supported by the securing mechanism must be described as well. 
   This protocol specification is intended for proposed standard or experimental, and must not assume any specific discovery mechanisms.

> 4. A specification describing an DNS-based emblem discovery mechanism (Proposed Standard or Experimental):

Author: I removed this, and the reason is that I think we expect binding and discovery to be related.
I added discovery to the binding PS (I collapsed to PS deliverables into one).
